x-pack/plugins/maps/public/classes/styles/vector/components/vector_style_editor.tsx

```diff
@@ -117,7 +117,10 @@ export class VectorStyleEditor extends Component<Props, State> {
       await this.props.layer.getStyleEditorFields()
     );
     const styleFields = styleFieldsHelper.getStyleFields();
-    if (!this._isMounted || _.isEqual(styleFields, this.state.styleFields)) {
+    if (
+      !this._isMounted ||
+      (_.isEqual(styleFields, this.state.styleFields) && this.state.styleFieldsHelper !== undefined)
```

Comment:
This was the regression. Editor did not render when `styleFieldsHelper` is undefined and `styleFieldsHelper` was not getting set when styleFields was an empty array since its initialized to an empty array.